src/core/useVideoTexture.tsx

```diff
@@ -19,11 +19,12 @@ export function useVideoTexture(src: string, props: Partial<VideoTextureProps>)
     ...props,
   }
   const gl = useThree((state) => state.gl)
-  const texture = suspend<[url: string], () => Promise<THREE.VideoTexture>>(
+  const texture = suspend<[urlOrStream: string | MediaProvider], () => Promise<THREE.VideoTexture>>(
     () =>
       new Promise((res, rej) => {
         const video = Object.assign(document.createElement('video'), {
-          src,
+          src: (typeof src === 'string' && src) || undefined,
+          srcObject: (src instanceof MediaStream && src) || undefined,
```

Comment:
If `src` is from `.getDisplayMedia()` set the `srcObject` of the video (rather the basic `src`)